src/appengine/handlers/cron/triage.py

```diff
@@ -250,11 +257,94 @@ def _check_and_update_similar_bug(testcase, issue_tracker):
   return False
 
 
-def _file_issue(testcase, issue_tracker):
+def _get_job_bugs_filing_max(job_type):
+  """"Gets the maximum number of bugs that can be filed for a given job."""
+  if job_type in _bug_filing_max_24_hours_per_job:
+    return _bug_filing_max_24_hours_per_job[job_type]
+
+  max_bugs = None
+  job = data_types.Job.query(data_types.Job.name == job_type).get()
+  if job and 'BUG_FILING_MAX_24_HOURS_PER_JOB' in job.get_environment():
+    try:
+      max_bugs = int(job.get_environment()['BUG_FILING_MAX_24_HOURS_PER_JOB'])
+    except Exception:
+      logs.log_error(
+          'Invalid environment value of \'BUG_FILING_MAX_24_HOURS_PER_JOB\' '
+          f'for job type {job_type}.')
+
+  _bug_filing_max_24_hours_per_job[job_type] = max_bugs
+  return max_bugs
+
+
+def _get_project_bugs_filing_max():
+  """Gets the maximum number of bugs that can be filed per project."""
+  global _bug_filing_max_24_hours_per_project
+  if not _bug_filing_max_24_hours_per_project:
+    config = local_config.IssueTrackerConfig()
+    try:
+      _bug_filing_max_24_hours_per_project = int(
+          config.get('BUG_FILING_MAX_24_HOURS_PER_PROJECT',
+                     BUG_FILING_MAX_24_HOURS_PER_PROJECT_DEFAULT))
+    except:
+      logs.log_error(
+          'Invalid config value of \'BUG_FILING_MAX_24_HOURS_PER_PROJECT\'')
+  return _bug_filing_max_24_hours_per_project
+
+
+def _throttle_bug(testcase, bug_filed_24_hours_per_job,
+                  bug_filed_24_hours_per_project):
+  """Returns whether the current bug needs to be throttled."""
+  job_bugs_filing_max = _get_job_bugs_filing_max(testcase.job_type)
+  valid_timestamp = datetime.datetime.now() - datetime.timedelta(hours=24)
+
+  # Check if the job type has a bug filing limit.
+  if job_bugs_filing_max:
+    # Get the number of bugs filed for the current job in the past 24 hours.
+    # First check the cache, then query the datastore if the value is not found.
+    count_per_job = bug_filed_24_hours_per_job.get(
```

Comment:
I think `count_per_job = bug_filed_24_hours_per_job.get(...) or query...` can save us a db query.
Also, moving this query to a function probably would make things a bit easy to read.